Extension/src/LanguageServer/configurations.ts

```diff
@@ -1343,9 +1356,28 @@ export class CppProperties {
             }
             curTextStartOffset = configStart + 1;
             curText = curText.substr(curTextStartOffset); // Remove earlier configs.
-            const nameEnd: number = curText.indexOf(":");
+            let nameEnd: number = curText.indexOf(":");
             curTextStartOffset += nameEnd + 1;
             curText = curText.substr(nameEnd + 1);
+            // Check if the conig name is unique, before removing later configs
+            let tailText: string = curText;
+            let tailTextStartOffset: number = curTextStartOffset;
+            let dupliacteNameStart: number = tailText.search(new RegExp(`{\\s*"name"\\s*:\\s*"${escapeStringRegExp(currentConfiguration.name)}"`));
```

Comment:
It looks like this RegExp is created twice.  Maybe create just one to reference from both places?